src/Message/Part.php

```diff
@@ -218,6 +219,11 @@ protected function parseStructure(\stdClass $structure)
             $this->type = self::TYPE_UNKNOWN;
         }
 
+        // fix strange encoding that more than ENCOTHER = 5
+        if ($structure->encoding > ENCOTHER) {
+            throw new MessageUnsupportedEncodeException($this->messageNumber, $this->partNumber);
```

Comment:
Implemented in https://github.com/ddeboer/imap/pull/197 thank you for your feedback